llvm/lib/Transforms/Scalar/Reassociate.cpp

```diff
@@ -2180,7 +2180,7 @@ void ReassociatePass::OptimizeInst(Instruction *I) {
   // is not further optimized, it is likely to be transformed back to a
   // short-circuited form for code gen, and the source order may have been
   // optimized for the most likely conditions.
-  if (I->getType()->isIntegerTy(1))
+  if (I->getType()->isIntOrIntVectorTy(1))
```

Comment:
With the extension to vector types, this comment needs an update. For vector types there's not going to be any short-circuiting.